catalogue_pipeline/ingestor/src/test/scala/uk/ac/wellcome/platform/ingestor/services/WorkIndexerTest.scala

```diff
@@ -83,20 +89,188 @@ class WorkIndexerTest
+              identifiedNewWork)
+          }
+        }
+      }
+    }
+    it("doesn't override a redirected Work with identified work same version") {
+      val redirectedWork = createIdentifiedRedirectedWorkWith(version = 3)
+      val identifiedWork = createIdentifiedWorkWith(
+        canonicalId = redirectedWork.canonicalId,
+        version = 3)
+
+      withLocalElasticsearchIndex { indexName =>
+        withWorkIndexer { workIndexer =>
+          val identifiedWorkInsertFuture = for {
+            _ <- workIndexer.indexWorks(
+              works = List(redirectedWork),
+              indexName = indexName,
+              documentType = documentType
+            )
+            result <- workIndexer.indexWorks(
+              works = List(identifiedWork),
+              indexName = indexName,
+              documentType = documentType
+            )
+          } yield result
+          whenReady(identifiedWorkInsertFuture) { result =>
+            // Give Elasticsearch enough time to ingest the work
+            Thread.sleep(700)
+            result.right.get should contain(identifiedWork)
+
+            assertElasticsearchEventuallyHasWork(
+              indexName = indexName,
+              redirectedWork)
+          }
+        }
+      }
+    }
+    it("override a identified Work with invisible work with higher version") {
+      val work = createIdentifiedWorkWith(version = 3)
+      val invisibleWork = createIdentifiedInvisibleWorkWith(
+        canonicalId = work.canonicalId,
+        version = 4)
+
+      withLocalElasticsearchIndex { indexName =>
+        withWorkIndexer { workIndexer =>
+          val invisibleWorkInsertFuture = for {
+            _ <- workIndexer.indexWorks(
+              works = List(work),
+              indexName = indexName,
+              documentType = documentType
+            )
+            result <- workIndexer.indexWorks(
+              works = List(invisibleWork),
+              indexName = indexName,
+              documentType = documentType
+            )
+          } yield result
+          whenReady(invisibleWorkInsertFuture) { result =>
+            // Give Elasticsearch enough time to ingest the work
+            Thread.sleep(700)
+            result.right.get should contain(invisibleWork)
+
+            assertElasticsearchEventuallyHasWork(
+              indexName = indexName,
+              invisibleWork)
+          }
+        }
+      }
+    }
+
+  }
 
   it("replaces a Work with the same version") {
     val work = createIdentifiedWorkWith(version = 3)
     val updatedWork = work.copy(title = "a different title")
 
     withLocalElasticsearchIndex { indexName =>
-      insertIntoElasticsearch(indexName = indexName, work)
-
       withWorkIndexer { workIndexer =>
-        val future = workIndexer.indexWorks(
-          works = List(updatedWork),
-          indexName = indexName,
-          documentType = documentType
-        )
+        val future = for {
```

Comment:
You can reuse the helper function here.

Reply:
which helper function?

Reply:
The helper function I suggested you should write further up in the patch. :-)

Reply:
Ah yes